### PR TITLE
build: add schematics to @angular/material package

### PR DESCRIPTION
**Pull request by jelbourn**
This also fixes a handful of minor issues with the blueprints

### CODE REVIEW

File: schematics/utils/devkit-utils/component.ts [hunk 1]
```diff
@@ -16,7 +16,6 @@ import {
 } from '@angular-devkit/schematics';
 import 'rxjs/add/operator/merge';
 import * as ts from 'typescript';
-import * as stringUtils from '@schematics/angular/strings';
 import { addDeclarationToModule, addExportToModule } from './ast-utils';
```
Comment (crisbeto) [inline schematics/utils/devkit-utils/component.ts]: Not something that was added in this PR, but the spaces between the curly braces are inconsistent with the rest of the project.

File: tools/gulp/tasks/material-release.ts [hunk 2]
```diff
@@ -25,21 +28,40 @@ const prebuiltThemeGlob = join(outputDir, '**/theming/prebuilt/*.css?(.map)');
...
- * a bundled SCSS file for theming
+ * Task that will build the material package. Special treatment for this package includes:
+ * - Copying all prebuilt themes into the package
+ * - Bundling theming scss into a single theming file
+ * - Copying schematics code into the packatge
  */
 task('material:prepare-release', sequenceTask(
-  'material:build',
-  ['material:copy-prebuilt-themes', 'material:bundle-theming-scss']
+  ['material:build', "schematics:build"],
```
Comment (crisbeto) [inline tools/gulp/tasks/material-release.ts]: `schematics:build` should be in single quotes.

File: tools/gulp/tasks/material-release.ts [hunk 3]
```diff
@@ -25,21 +28,40 @@ const prebuiltThemeGlob = join(outputDir, '**/theming/prebuilt/*.css?(.map)');
...
+  ['material:copy-prebuilt-themes', 'material:bundle-theming-scss', 'material:copy-schematics'],
 ));
 
+task('material:copy-schematics', () => {
+  src(schematicsGlobs).pipe(dest(join(join(releasePath, 'schematics'))));
```
Comment (crisbeto) [inline tools/gulp/tasks/material-release.ts]: The second `join` around the first `join` seems unnecessary.

File: schematics/utils/devkit-utils/component.ts (hunk 1 above)
Comment (jelbourn) [reply schematics/utils/devkit-utils/component.ts]: Done

File: tools/gulp/tasks/material-release.ts (hunk 2 above)
Comment (jelbourn) [reply tools/gulp/tasks/material-release.ts]: Done

File: tools/gulp/tasks/material-release.ts (hunk 3 above)
Comment (jelbourn) [reply tools/gulp/tasks/material-release.ts]: Done

**Pull request merged**